invenio_communities/communities/permissions.py

```diff
@@ -1,106 +1,107 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2020 CERN.
+# Copyright (C) 2020-2021 CERN.
 #
 # Invenio is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """Community permissions."""
 
+from elasticsearch_dsl.query import Q
 from flask_principal import UserNeed
-from flask_security import current_user
 from invenio_access.permissions import any_user
-from invenio_records_permissions.generators import AnyUser, Generator
+from invenio_rdm_records.services.generators import \
+    IfRestricted as IfRestrictedBase
+from invenio_records_permissions.generators import AnyUser, \
+    AuthenticatedUser, Generator
 from invenio_records_permissions.policies import BasePermissionPolicy
 
-# from invenio_communities.members.permissions import CommunityMember
 
-
-class CommunityVisibilityPolicy(Generator):
-    """Allows only community members to read the community if its hidden."""
-
-    def needs(self, community=None, **kwargs):
-        """."""
-        if community['visibility'] == 'public':
-            return [any_user]
+class IfRestricted(IfRestrictedBase):
```

Comment:
Shelved: As discussed IRL, since we only need to override the `access.visibility.{private,public}` variable here, this could be extracted or made uniform with theoriginal class.